CONTRIBUTING.md

```diff
@@ -1,8 +1,123 @@
+    - If the changes you wish to make to PyPose involve significant modifications, it is recommended that you create an issue first. Within the issue, describe the error and information on how to reproduce the bug. Other developers will then review the issue and engage in discussion.
+    - If you require additional information regarding a specific issue, please do not hesitate to request it, and we will gladly provide it to you.
+2. If you have an idea for a new feature in PyPose and wish to implement it, we recommend that you create an issue to post about your proposed feature. From there, we can discuss the design and implementation. Once we agree that the plan looks good, go ahead and implement it.
 
-## 1.1 Build docs locally
+Once you implement and test your feature or bug-fix, please submit a Pull Request. 
+
+## Workflow
+This document covers a step-by-step guide of contributing. It will tell you how to start from setting up the code repository and create a Pull Request step by step. This guide is modified from [OpenMMlab](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md)'s contribution guide.
+
+### 1. Fork and clone
+
+If you are posting a pull request for the first time, you should fork the PyPose repository by clicking the **Fork** button in the top right corner of the GitHub page, and then the forked repositories will appear under your GitHub profile.
+
+Then, you can clone the repositories to local:
+
+```shell
+git@github.com:{your_github_id}/pypose.git
+```
+
+After that, you should add official repository as the upstream repository
+
+```bash
+git remote add upstream git@github.com:pypose/pypose.git
+```
+
+Check whether remote repository has been added successfully by `git remote -v`
+
+```bash
+origin	git@github.com:{your_github_id}/pypose.git (fetch)
+origin	git@github.com:{your_github_id}/pypose.git (push)
+upstream	git@github.com:pypose/pypose.git (fetch)
+upstream	git@github.com:pypose/pypose.git (push)
+```
+
+> Here's a brief introduction to origin and upstream. When we use "git clone", we create an "origin" remote by default, which points to the repository cloned from. In the above example, "origin" is your own forked repo. As for "upstream", we add it ourselves to point to the target repository, which is PyPose's official repo `pypose/pypose`. Of course, you could name it to any arbitrary name other than `upstream`. Usually, contributors will push the code to "origin". If the pushed code conflicts with the latest code in official("upstream"), the contributor should pull the latest code from upstream to resolve the conflicts, and then push to "origin" again. The posted Pull Request will be updated automatically.
+
+### 2. Create a development branch
+After cloning a local copy of the repo, we should create a branch based on the master branch to develop the new feature or fix the bug. The proposed branch name is `username/pr_name`
+
+```shell
+git checkout -b {your_github_id}/{pr_name}
```

Comment:
The developers don't have to use a development branch within the forked repo. They can use their own repo's `main` branch and create a pull request from that.

Reply:
Is it necessary to keep the step, to prevent people who are in the PyPose organization accidentally push to `pypose/main`?

Reply:
Oh, they won't succeed, the current settings need at least 1 reviewer approval to merge a pull request and no one can directly push to `main` branch.
but yes, for people in PyPose Github organization, they should use a `dev` branch.